druid/src/env.rs

```diff
@@ -133,6 +133,35 @@ pub enum KeyOrValue<T> {
     Key(Key<T>),
 }
 
+/// A trait for anything that can resolve a value of some type from the [`Env`].
+///
+/// This is a generalization of the idea of [`KeyOrValue`], mostly motivated
+/// by wanting to improve the API used for checking if items in the [`Env`] have changed.
+///
+/// [`Env`]: struct.Env.html
+/// [`KeyOrValue`]: enum.KeyOrValue.html
+pub trait KeyLike<T> {
+    /// Returns `true` if this item has changed between the old and new [`Env`].
+    ///
+    /// [`Env`]: struct.Env.html
+    fn changed(&self, old: &Env, new: &Env) -> bool;
+}
+
+impl<T: ValueType> KeyLike<T> for Key<T> {
+    fn changed(&self, old: &Env, new: &Env) -> bool {
+        !old.get_untyped(self).same(new.get_untyped(self))
```

Comment:
I've got a panic on this line.
I'm not sure if I am doing anything wrong, but I think this would be better written as:
```suggestion
        match (old.try_get_untyped(self), new.try_get_untyped(self)) {
            (Err(_), Err(_)) => false,
            (Err(_), Ok(_)) => true,
            (Ok(_), Err(_)) => true,
            (Ok(old_value), Ok(new_value)) => !old_value.same(new_value),
        }
```